tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/EtcTypeChecker2.scala

```diff
@@ -0,0 +1,277 @@
+package at.forsyte.apalache.tla.typecheck.etc
+
+import at.forsyte.apalache.tla.typecheck._
+import at.forsyte.apalache.tla.typecheck.etc.EtcTypeChecker2.UnwindException
+
+/**
+  * ETC: Embarrassingly simple Type Checker.
+  *
+  * @author Igor Konnov
+  */
+class EtcTypeChecker2(varPool: TypeVarPool) extends TypeChecker with EtcBuilder {
```

Comment:
There is no non-obvious reason. Similar to test fixtures and logging, it was just easier to have the constructors in `EtcBuilder`.

Reply:
Just to clarify: my question is not why do we keep the constructors in `EtcBuilder`: that seems like a sensible grouping. My question is why we mix in the `EtcBuilder` class instead of just calling, e.g., `EtcBuilder.mkConst` etc.

Reply:
If there's not a reason, seems like we can simplify the class hierarchy here without any additional cost?

Reply:
Well, mixtures are just a common idiom in scala. For instance, see `LazyLogging`. Their instances are not meant to be used individually, but they mix-in useful behavior. If you worry about complexity of the class hierarchy, we should inherit from `LazyLogging` with the `protected` level. That needs some googling thought :-)

Reply:
Ah I see. Thanks for explaining!